### PR TITLE
Convert bash scripts to use '#!/usr/bin/env bash' shebang line

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

Convert bash scripts to use `#!/usr/bin/env bash` shebang line and add a check to make sure all executables in allow list exist.

Added all shell scripts to allow list, and changed them to be executable.

Modified the analyze script to enforce that the shell after `/usr/bin/env` is either `bash` or `sh` if the filename ends in `.sh`, and removed the shebang shell enforcement from the license checking script.

### CODE REVIEW

File: analyze/analyze.dart [hunk 1]
```diff
@@ -256,6 +290,13 @@ Future<void> _checkForNewExecutables() async {
       'must add this file to kExecutableAllowlist in analyze/analyze.dart',
     );
   }
+  if (unexpectedShebangShellCount > 0) {
+    throw Exception(
+      'found $unexpectedShebangShellCount unexpected shell #! line'
+      '${unexpectedShebangShellCount == 1 ? '' : 's'}! If this was intended, you '
+      'must modify kShebangRegex in analyze/analyze.dart',
+    );
+  }
```
Comment (keyonghan) [inline analyze/analyze.dart:299]: This dups the above lines.

File: analyze/analyze.dart [hunk 2]
```diff
@@ -217,37 +217,71 @@ Future<EvalResult> _evalCommand(
...
+    final bool inAllowList = kExecutableAllowlist.contains(relativePath);
+    if (isExecutable && !inAllowList) {
       unexpectedExecutableCount += 1;
       print('$relativePath is executable: ${(stat.mode & 0x1FF).toRadixString(2)}');
     }
+    if (inAllowList && file.path.endsWith('.sh')) {
+      final String shebang = file.readAsLinesSync().first;
+      if (!shebang.startsWith(RegExp(kShebangRegex))) {
+        unexpectedShebangShellCount += 1;
+        print("$relativePath is has the initial line of $shebang, which doesn't match '$kShebangRegex'");
```
Comment (keyonghan) [inline analyze/analyze.dart:282]:
```suggestion
        print("$relativePath has the initial line of $shebang, which doesn't match '$kShebangRegex'");
```

File: analyze/analyze.dart (hunk 1 above)
Comment (gspencergoog) [reply analyze/analyze.dart:299]: No, it doesn't.  It's a different message.

Comment (keyonghan) [reply analyze/analyze.dart:299]: Oops, my bad.

**Review by keyonghan — APPROVED** [review]
Thanks for enforcing the license check in analyzer. LGTM!

**Comment by keyonghan** [comment]
The newly added build.sh needs an update: https://github.com/gspencergoog/cocoon/blob/1698c8819879acca527643cfb9cd621d4969ba34/cipd_packages/doxygen/tool/build.sh#L1

**Comment by gspencergoog** [comment]
> The newly added build.sh needs an update: [gspencergoog/cocoon@1698c88/cipd_packages/doxygen/tool/build.sh#L1](https://github.com/gspencergoog/cocoon/blob/1698c8819879acca527643cfb9cd621d4969ba34/cipd_packages/doxygen/tool/build.sh#L1)

At least we know the new check works!

**Pull request merged**